lib/member_page.rb

```diff
@@ -0,0 +1,112 @@
+require 'scraped_page'
+
+class String
+  def tidy
+    gsub(/[[:space:]]+/, ' ').strip
+  end
+end
+
+class MemberPage < ScrapedPage
+  field :source do
+    url
+  end
+
+  field :image do
+    noko.at_css('.dep_img img/@src').text.tidy
+  end
+
+  field :name do
+    name_and_id.split('-')[0..-2]
+               .reject { |w| w == 'ru' }
+               .map(&:upcase)
+               .join(' ')
+  end
+
+  field :name_ru do
+    noko.at_css('.dep_info h1').text.tidy
+  end
+
+  field :id do
+    name_and_id.split('-').last
+  end
+
+  field :area do
+    area_and_id[0]
+  end
+
+  field :area_id do
+    area_and_id[1]
+  end
+
+  field :telephone do
+    # The markdown within the target div is non-semantic and inconsistent
+    # from page to page. Splitting by preceding characters does not work
+    # because preceding characters vary.
+    # Any phone number in one of the two telephone formats on the site
+    # ( (x-xxxx) xx-xx-xx & (+xxxxx) xxx-xx-xx ) will be matched by the regex.
+    # Multiple numbers will be joined by a semi-colon.
+    contact_nodes.text
+                 .scan(/\(?\+?[0-9]+-?[0-9]+\)\s[0-9]+-[0-9]+-[0-9]+/)
+                 .map(&:tidy)
+                 .join(';')
+  end
+
+  field :website do
+    # The markdown within the target div is non-semantic and inconsistent
+    # from page to page. Splitting by preceding characters does not work
+    # because preceding characters vary.
+    # The regex makes the naive assumption that any string begining with
+    # http:// followed by a sequence of dots and letters is a web address.
```

Comment:
This comment doesn't match what you're actually scanning for (e.g. you include digits in the regex). Trying to build a regex to match a URL is close to madness though. If you really need a regex for this, there are plausible ones online, but those are overkill for this. If you're only looking for strings starting with "http://", then you don't need a regex for that.